File: tests/go/cases/case_108_local_cache_named_retained_01.go
package escape_tests

var retainedCase108 = []map[string]string{}

func Case108LocalCacheNamedRetained01(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "local_cache_named_retained_01",
		"entity": "stress",
		"stage": "evaluation",
		"input": raw,
		"checksum": "local_cache_named_retained_01:" + raw,
	}
	retainedLocal := []map[string]string{}
	retainedLocal = append(retainedLocal, payload)
	// SAFE: local slice dies at return.
	return payload["checksum"]
}

File: tests/nodejs/cases/case_108_local_cache_named_retained_01.js
/**
 * Task case 108: local_cache_named_retained_01 false-positive/false-negative stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case108LocalCacheNamedRetained01(input) {
  const taskName = 'local_cache_named_retained_01';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'stress',
    stage: 'evaluation',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  const retainedCacheLocal = new Map();
  retainedCacheLocal.set('tmp', payload);
  // SAFE: local map does not escape.
  return payload.checksum;
}

module.exports = {
  case108LocalCacheNamedRetained01,
};
